v2/git-initializer/main_test.go

```diff
@@ -0,0 +1,211 @@
+// +build integration
+
+package main
+
+import (
+	"context"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/brigadecore/brigade/sdk/v2"
+	"github.com/brigadecore/brigade/sdk/v2/authx"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+)
+
+type testcase struct {
+	name          string
+	project       core.Project
+	terminalPhase core.WorkerPhase
+}
+
+var defaultBrigadeJS = `
+	const { events } = require("brigadier");
+
+	events.on("github.com/brigadecore/brigade/cli", "exec", () => {
+		console.log("Hello, World!");
+	});
+`
+
+var testcases = []testcase{
+	{
+		name: "GitHub - Public Repo - https",
+		project: core.Project{
+			ObjectMeta: meta.ObjectMeta{
+				ID: "public-https",
+			},
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						CloneURL: "https://github.com/krancour/brigade2-pipeline-demo.git",
+					},
+					DefaultConfigFiles: map[string]string{
+						"brigade.js": defaultBrigadeJS,
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	// TODO: once we have mech for injecting private ssh key in secure manner
```

Comment:
Can it be a normal secret?

I think the missing mech you're talking about is the ability to define secrets declaratively?

In the meantime, we could add an optional "post project create" (terrible name) function to each test case. It can receive an API client as an argument and complete any setup-- which could include setting a secret.

Example:

```
testCases := []struct{
  project sdk.Project,
  postProjectCreate func(sdk.APIClient),
} {
  {
    project: ...,
    postProjectCreate: func(client sdk.APIClient) {
      err := client.Core().Project().Secrets().Set(...)
      require.NoErr(t, err)
    },
  },
  // ...
}
// ...
for _, testCase := range testCases {
  _, err := client.Core().Projects().Create(testCase.project)
  require.NoErr(t, err)
  if testCase.postProjectCreate != nil {
    testCase.postProjectCreate(client)
  }
}
```

Note example above was off the top of my head-- I didn't try it.

Reply:
Ah, great idea!  Indeed, I'm looking for a mech like that -- but the trickier bit is: What's a good approach for adding an ssh key that is valid for the private repo used in this test case.  There's setup involved: repo needs to exist, the public key needs to be added, the private key needs to be wired in to the test project as sketched above.  We'd want to support any dev running these locally and also in CI... 🤔

Reply:
Ah..... I completely forgot that such a server would need to be set up. 🤦 

Follow up PR then?